### PR TITLE
Improved Sonar coverage.

### PR DESCRIPTION
**Pull request by valean27**
Fix sonar issues, added unit tests in order to increase coverage

### CODE REVIEW

File: src/test/java/org/mule/extension/hdfs/automation/unit/ConnectionTest.java [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+        expectedException.expectMessage("\"nameNodeUri\" can not be null or empty");
+        hadoopConfigurationUtil.getKerberosAuthConfig(null,null,null,null);
+
+    }
+    @Test
+    public void testKerberosWithNullPrincipal()
+    {
+        expectedException.expect(IllegalArgumentException.class);
+        expectedException.expectMessage("\"principal\" can not be null or empty.");
+        hadoopConfigurationUtil.getKerberosAuthConfig("DA",null,null,null);
```
Comment (github-ccampian) [inline src/test/java/org/mule/extension/hdfs/automation/unit/ConnectionTest.java]: Remove "DA" and use something more appropriate in this context.

File: src/test/java/org/mule/extension/hdfs/automation/unit/HdfsOperationsTest.java [hunk 2]
```diff
@@ -36,14 +45,31 @@
     public void testDeleteFile() {
         try {
             when(fileSystem.delete(any(Path.class), eq(false))).thenReturn(true);
-
             hdfsOperations.deleteFile(hdfsConnection, "zoo");
```
Comment (github-ccampian) [inline src/test/java/org/mule/extension/hdfs/automation/unit/HdfsOperationsTest.java]: Remove "zoo" and use a more appropriate word.

File: src/test/java/org/mule/extension/hdfs/automation/unit/HdfsOperationsTest.java [hunk 3]
```diff
@@ -36,14 +45,31 @@
...
             fail(e.getMessage());
         }
     }
 
     @Test
+    public void testUnsuccesfullyDeleteFile() throws IOException {
+            expectedException.expect(ModuleException.class);
+            expectedException.expectMessage(StringContains.containsString("Unable to delete file!"));
+            when(fileSystem.delete(any(Path.class), eq(false))).thenReturn(false);
+            hdfsOperations.deleteFile(hdfsConnection,"fdsdfs");
```
Comment (github-ccampian) [inline src/test/java/org/mule/extension/hdfs/automation/unit/HdfsOperationsTest.java]: Remove "fdsdfs" and use a more appropriate word.

File: src/test/java/org/mule/extension/hdfs/automation/unit/HdfsOperationsTest.java [hunk 4]
```diff
@@ -75,6 +128,26 @@ public void testRename() {
...
         }
     }
+    @Test
+    public void testUnsuccesfullyRename() throws IOException {
+
+            expectedException.expect(ModuleException.class);
+            expectedException.expectMessage(StringContains.containsString("Unable to rename path!"));
+
+            when(fileSystem.rename(any(Path.class), any(Path.class))).thenReturn(false);
+            hdfsOperations.rename(hdfsConnection, "foo", "zoo");
```
Comment (github-ccampian) [inline src/test/java/org/mule/extension/hdfs/automation/unit/HdfsOperationsTest.java]: Remove "zoo" and use a more appropriate word.

File: src/test/java/org/mule/extension/hdfs/automation/unit/HdfsOperationsTest.java [hunk 5]
```diff
@@ -130,6 +245,16 @@ public void testSetPermission() {
     }
 
     @Test
+    public void testUnsuccesfullySetPermission() {
+        try {
+            expectedException.expectMessage(StringContains.containsString("Invalid request data :Can not create a Path from a null string"));
```
Comment (github-ccampian) [inline src/test/java/org/mule/extension/hdfs/automation/unit/HdfsOperationsTest.java]: Create a constant with 'invalid request data :Can not create a Path from a null string' seems to be used in more places.

**Comment by github-ccampian** [comment]
+1

**Comment by danpece** [comment]
+1

**Pull request merged**